src/types/utils.cpp

```diff
@@ -21,34 +24,55 @@ static constexpr bool _isNumber(const wchar_t wch) noexcept
     return wch >= L'0' && wch <= L'9'; // 0x30 - 0x39
 }
 
-// Function Description:
-// - Creates a String representation of a guid, in the format
-//      "{12345678-ABCD-EF12-3456-7890ABCDEF12}"
-// Arguments:
-// - guid: the GUID to create the string for
-// Return Value:
-// - a string representation of the GUID. On failure, throws E_INVALIDARG.
-std::wstring Utils::GuidToString(const GUID guid)
+[[gsl::suppress(bounds)]]
+static std::wstring guidToStringCommon(const GUID& guid, size_t offset, size_t length)
 {
-    return wil::str_printf<std::wstring>(L"{%08x-%04x-%04x-%02x%02x-%02x%02x%02x%02x%02x%02x}", guid.Data1, guid.Data2, guid.Data3, guid.Data4[0], guid.Data4[1], guid.Data4[2], guid.Data4[3], guid.Data4[4], guid.Data4[5], guid.Data4[6], guid.Data4[7]);

```

Comment:
BTW the documentation doesn't match the implementation here as `%02x` results in lowercase hexadecimal. I've switched this code over to the combaseapi functions now, as I felt like they're more "authoritative" when it comes to GUIDs.

Reply:
Ah, I see now. The unit tests even asserted on the lowercase characters. So it was indeed just the documentation that was incorrect. I'll change the unit tests because of the 2nd point above, but I'd be happy to change everything back to a plain `sprintf`.

Reply:
Actually, thinking about it, the entire project uses lowercase UUIDs already, so it'd be weird if this didn't. I'll change it back.